src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,171 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private final int defaultCapacity = 16;
+    private final float loadFactor = 0.75f;
+    private Node<K, V>[] tables;
+    private int threshold;
+    private int size = 0;
+
+    public MyHashMap() {
+        tables = new Node[defaultCapacity];
+        threshold = (int) (loadFactor * tables.length);
+    }
 
     @Override
     public void put(K key, V value) {
+        Node<K, V> newNode = null;
+        int index = index(key);
+        Objects.checkIndex(index, tables.length);
+        if (key == null) {
+            key = (K) Integer.valueOf(index);
+        }
```

Comment:
Are you sure you need this code? If key==null, the index() method should return 0

Reply:
it was crushed to reduce writing code...
i.e., I set the condition to zero, then put in index zero
and so I simplified zero, converted the integer
 to zero and immediately put it as needed ...

Reply:
you dont need this if, remove it

this code doesn't even make any sense. You are trying to cast Integer to K type? What if K type is some class like Animal?

if key is null it should stay null - HashMap allows storing of null keys